static/src/javascripts/projects/common/modules/ui/subscription-banners/guardian-weekly-banner.spec.js

```diff
@@ -0,0 +1,293 @@
+// @flow
+
+import { bannerTemplate as gwBannerTemplate } from './guardian-weekly-banner-template';
+import { selectorName, createBannerShow } from './subscription-banner';
+
+jest.mock('ophan/ng', () => null);
+jest.mock('common/modules/commercial/user-features.js', () => jest.fn());
+jest.mock('common/modules/commercial/contributions-utilities.js', () => ({
+    getReaderRevenueRegion: jest.fn().mockReturnValue('united-kingdom'),
+}));
+jest.mock('common/modules/identity/api.js', () => ({
+    isUserLoggedIn: jest.fn().mockReturnValue(true),
+}));
+jest.mock('lib/report-error', () => jest.fn());
+
+const removeAllSpaces = (str: string) => str.replace(/\s/g, '');
+
+describe('Guardian Weekly Banner', () => {
+    const bannerSelector = selectorName('subscription-banner');
+    const mockTracking = {
+        signInUrl: 'signInUrl',
+        gaTracking: jest.fn(),
+        subscriptionUrl: 'subscriptionUrl',
+        trackBannerView: jest.fn(),
+        trackBannerClick: jest.fn(),
+        trackCloseButtons: jest.fn(),
+    };
+
+    afterEach(() => {
+        if (document.body) {
+            document.body.innerHTML = '';
+        }
+    });
+
+    it('Should append a Guardian Weekly Banner to the document', () => {
+        let appendedBanner = 'Banner Not Appended';
+
+        const show = createBannerShow(mockTracking, gwBannerTemplate, false);
+        show();
+
+        const banner = gwBannerTemplate(
+            mockTracking.subscriptionUrl,
+            mockTracking.signInUrl,
+            false
+        );
+
+        if (document.body) {
+            appendedBanner = document.body.innerHTML;
+        }
+
+        // removeAllSpaces is used because toEqual compares all tabs, spaces and new lines
+        expect(removeAllSpaces(banner)).toEqual(
+            removeAllSpaces(appendedBanner)
+        );
+    });
+
+    describe('Button Clicks', () => {
+        beforeEach(() => {
+            const show = createBannerShow(
+                mockTracking,
+                gwBannerTemplate,
+                false
+            );
+
+            show();
+        });
+
+        it('Should remove banner when close button is clicked', () => {
+            let appendedBanner = 'Banner Not Appended';
```

Comment:
Can we avoid using `let` here?
Maybe like this:
```
const appendedBanner = document.body ? document.body.innerHTML : ''
```

Reply:
Similar improvements can be done in the file `subscription-banner.spec.js` below.